i2c/oled_i2c/oled_i2c.c

```diff
@@ -99,17 +98,17 @@ void print_buf_page(uint8_t buf[], uint8_t page) {
 
 void print_buf_pages(uint8_t buf[]) {
     // prints all pages of a full length buffer
-    for (int i = 0; i < OLED_PAGES_NUM; i++) {
+    for (int i = 0; i < OLED_NUM_PAGES; i++) {
         printf("--page %d--\n", i);
         print_buf_page(buf, i);
     }
 }
 
 void print_buf_area(uint8_t buf[], struct render_area* area) {
     // print a render area of generic size
-    uint8_t area_width = area->end_col - area->start_page + 1;
+    uint8_t area_width = area->end_col - area->start_col + 1;
     uint8_t area_height = area->end_page - area->start_page + 1; // in pages, not pixels
-    for (int i = 0; i < area_height; i++) {
+    for (int i = area->start_page; i < area_height; i++) {
```

Comment:
Should this be `i <= area->end_page;` ?

Similarly, perhaps the
```c
for (int k = 0; k < area_width; k++) {
```
should be 
```c
for (int k = area->start_col; k <= area->end_col; k++) {
```
?
(although off the top of my head I'm not sure what the bit inside `buf[  ]` would need to become!)

Reply:
Hmmm, actually I think we should switch it back to 0. The reason is that there's a difference between printing just a little area (like the raspberry) and seeing what the entire frame (128x32) buffer has. If we print a little area, we just want to see the area and so place it in the left hand corner, so effectively start_col = 0 and start_page = 0. This also just makes things easier for the bit inside buf[] as we don't have to subtract any offsets, if, for example start_col is not 0. We really only the need the height and width of the area, but I wanted to make the "print" and "render" functions consistent and use the `render_area` struct.

Reply:
IMHO if you're going to change the function to always print from the top-left corner, then you ought to make that explicit by passing in separate `width` and `height` parameters, rather than passing in a `render_area` struct and just ignoring its `start_page` and `start_col` values.
But it's probably worth waiting to see what @kilograham says, as he knows the "feel" of the SDK and Examples much better than I do :slightly_smiling_face: